### PR TITLE
patching https://github.com/michaelklishin/pantomime/issues/17

### LINKED ISSUE #17: Expose type inheritance feature
I was looking for functionality similar to other shared-mime-info interfaces that will tell you if a type is a subtype of some other type, e.g. that `application/vnd.openxmlformats-officedocument.spreadsheetml.sheet` is an `application/zip`. This is exposed in [`org.apache.tika.mime.MediaTypeRegistry`](http://tika.apache.org/1.12/api/org/apache/tika/mime/MediaTypeRegistry.html).

I have already implemented a patch. If you're cool with PRs I can fork this project and do one. I have also included a protocol for coercing objects between `MimeType` and `MediaType`.

(Why does Tika have two separate, completely unrelated classes to represent type objects, anyway?)

### PR DESCRIPTION
**Pull request by doriantaylor**
Closes #17.

### CODE REVIEW

File: src/clojure/pantomime/mime.clj [hunk 1]
```diff
@@ -84,3 +86,34 @@
...
+  MimeType
+  (as-mime-type  [^MimeType x] x)
+  (as-media-type [^MimeType x] (.getType x))
+  MediaType
+  (as-mime-type  [^MediaType x] (for-name (.toString (.getBaseType x))))
+  (as-media-type [^MediaType x] x)
+)
+
+;; don't collide with 'core/instance?'
+(defn type-instance?
```
Comment (michaelklishin) [inline src/clojure/pantomime/mime.clj]: I'd use `instance-of?`. For some reason "type-instance?" doesn't make me think of MIME types and inheritance.

Comment (doriantaylor) [inline src/clojure/pantomime/mime.clj]: It's your baby!

File: src/clojure/pantomime/mime.clj [hunk 2]
```diff
@@ -84,3 +86,34 @@
...
+  MimeType
+  (as-mime-type  [^MimeType x] x)
+  (as-media-type [^MimeType x] (.getType x))
+  MediaType
+  (as-mime-type  [^MediaType x] (for-name (.toString (.getBaseType x))))
+  (as-media-type [^MediaType x] x)
+)
+
+;; don't collide with 'core/instance?'
+(defn instance-of?
```
Comment (doriantaylor) [inline src/clojure/pantomime/mime.clj]: okay there you go

**Comment by michaelklishin** [comment]
Thank you!

**Comment by doriantaylor** [comment]
thank you for merging!

**Pull request merged**